src/relations/mysql_provider.py

```diff
@@ -159,17 +159,26 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             self.database.set_endpoints(relation_id, f"{primary_endpoint}:3306")
             replicas_endpoint = socket.getfqdn(f"{self.charm.app.name}-replicas")
             self.database.set_read_only_endpoints(relation_id, f"{replicas_endpoint}:3306")
-            # TODO:
-            # add setup of tls, tls_ca and status
-            # add extra roles parsing from relation data
-            self.charm._mysql.create_application_database_and_scoped_user(
-                db_name, db_user, db_pass, "%"
-            )
 
             if "mysqlrouter" in extra_user_roles:
+                self.charm._mysql.create_application_database_and_scoped_user(
+                    db_name,
+                    db_user,
+                    db_pass,
+                    "%",
+                    # MySQL Router charm does not need a new database
+                    create_database=False,
+                )
                 self.charm._mysql.grant_privileges_to_user(
                     db_user, "%", ["ALL PRIVILEGES"], with_grant_option=True
                 )
+            else:
+                # TODO:
+                # add setup of tls, tls_ca and status
+                # add extra roles parsing from relation data
```

Comment:
The same question as here: https://github.com/canonical/mysql-operator/pull/210/files/d65a107bedb54fb4e57acd82c75cf90016163a00#r1194869066